apps/EchoServerClient/UdpEchoClient.v3

```diff
@@ -0,0 +1,93 @@
+// Copyright 2022 Virgil Authors. All rights reserved.
+// See LICENSE for details of Apache 2.0 license.
+
+// This implements a simple UDP echo client that connects to 127.0.0.1:5555 or
+// ::1:5555 and sends the message provided by the user from the commandline.
+
+def EXIT_FAILURE = -1;
+def options = BasicOptions.new(null);
+def help    = options.newBoolOption("h", false);
+def v6      = options.newBoolOption("v6", false);
+def connect = options.newBoolOption("connect", false);
+
+def usage() {
+	System.puts("Usage: UdpEchoClient [-h] [-v6] [-connect] \"<message string>\"\n");
+	System.puts("  -h        Print this option summary\n");
+	System.puts("  -v6       Use IPv6 instead of IPv4\n");
+	System.puts("  -connect  Use connected sockets instead of connectionless ones\n");
+	RiOs.exit(EXIT_FAILURE);
+}
+
+def main(args: Array<string>) {
+	args = options.parse(args);
+	if (help.get() || args.length == 0) {
+		usage();
+	}
+
+	var sock: Socket;
+	var packet: UdpPacket = null;
+	def localhost = if(v6.get(), Net.LOCALHOST_V6, Net.LOCALHOST_V4);
+
+	if (connect.get()) {
+		sock = ClientSocket.new(localhost, 5555, SocketType.Udp);
+	} else {
+		sock = UdpSocket.new(0, v6.get());
+		packet = UdpPacket.new(localhost, 5555);
+	}
+
+	def buffer = Array<byte>.new(512);
+	def strBldr = StringBuilder.new();
+
+	if (args.length == 0 || args.length > 1) {
+		usage();
+	}
+
+	strBldr.put1("Hello! Socket opened on fd %d\n", sock.fd);
```

Comment:
Here too, but since there is already a `buffer`, I might call the byte array the `iobuf` or something.